go-controller/pkg/crd/egressfirewall/v1/types.go

```diff
@@ -67,11 +67,14 @@ type EgressFirewallPort struct {
 
 // EgressFirewallDestination is the endpoint that traffic is either allowed or denied to
 type EgressFirewallDestination struct {
-	// cidrSelector is the CIDR range to allow/deny traffic to. If this is set, dnsName must be unset.
+	// cidrSelector is the CIDR range to allow/deny traffic to. If this is set, dnsName and nodeSelector must be unset.
 	CIDRSelector string `json:"cidrSelector,omitempty"`
-	// dnsName is the domain name to allow/deny traffic to. If this is set, cidrSelector must be unset.
+	// dnsName is the domain name to allow/deny traffic to. If this is set, cidrSelector and nodeSelector must be unset.
 	// +kubebuilder:validation:Pattern=^([A-Za-z0-9-]+\.)*[A-Za-z0-9-]+\.?$
 	DNSName string `json:"dnsName,omitempty"`
+	// nodeSelector will allow/deny traffic to the Kubernetes node IP of selected nodes. If this is, set
+	// cidrSelector and DNSName must be unset.
```

Comment:
how is this condition validated? I havent seen it so far. Are we rejecting when this condition isnt true otherwise the user will see unexpected results.

Reply:
do u need `// +optional` or this new field , also indicate empty selector "Default" allow/deny to any node

Reply:
will add @msherif1234 

@martinkennelly I wasn't really validating it, just ignoring nodeSelector if one of the other 2 were set. This is how the previous egress firewall interacted between DNSName and CIDRSelector

Reply:
The order right now is:
1. If dns name provided, use that
2. else if cidr , use that
3. else use node selector

I could add some validation to make sure only 1 is set. Seems like a good idea to me. wdyt?